### PR TITLE
fix nightly tests

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: linker/src/lib.rs [hunk 1]
```diff
@@ -879,7 +879,7 @@ namespace main_vm(64..128);
     fn permutation_instructions() {
         let expected = r#"namespace main(128);
     let _operation_id;
-    query |__i| std::prover::provide_if_unknown(_operation_id, __i, || 13);
+    query |__i| std::prover::provide_if_unknown(_operation_id, __i, || 9);
```
Comment (leonardoalt) [inline linker/src/lib.rs:882]: these were caused by changing the asm test

File: pipeline/src/test_util.rs [hunk 2]
```diff
@@ -265,7 +265,14 @@ pub fn gen_halo2_proof(pipeline: Pipeline<Bn254Field>, backend: BackendVariant)
...
     let setup_file_path = output_dir.join("params.bin");
-    let max_degree = pil.degrees().into_iter().max().unwrap();
+    let max_degree = pil
+        .degree_ranges()
+        .into_iter()
+        .map(|range| range.max)
+        .collect::<Vec<_>>()
+        .into_iter()
+        .max()
+        .unwrap();
```
Comment (leonardoalt) [inline pipeline/src/test_util.rs:275]: this function is called for both Composite and Mono proofs, and it assumed all degrees were unique ranges before. This now takes the max of all max degrees for the halo2 setup

File: test_data/asm/permutations/block_to_block.asm [hunk 3]
```diff
@@ -43,9 +43,7 @@ machine Main with degree: 256 {
     Binary4 bin;
     Binary4x bin4;
 
-    // two permutations to machine bin
     instr or X, Y -> Z link ~> Z = bin.or(X, Y);
-    instr or_into_B X, Y link ~> B' = bin.or(X, Y);
```
Comment (leonardoalt) [inline test_data/asm/permutations/block_to_block.asm:48]: unfortunately Halo2 has a bug somewhere and we can't do this.
This is testes in other places (including RISCV tests) with witgen

File: test_data/asm/permutations/vm_to_block.asm [hunk 4]
```diff
@@ -11,9 +11,7 @@ machine Main with degree: 128 {
 
     Binary4 bin;
 
-    // two permutations to machine bin
     instr or X, Y -> Z link ~> Z = bin.or(X, Y);
-    instr or_into_B X, Y link ~> B' = bin.or(X, Y);
```
Comment (leonardoalt) [inline test_data/asm/permutations/vm_to_block.asm:16]: same here re halo2

File: test_data/asm/permutations/block_to_block.asm (hunk 3 above)
Comment (pacheco) [reply test_data/asm/permutations/block_to_block.asm:48]: is it possible to disable halo2 for the test instead?

Comment (leonardoalt) [reply test_data/asm/permutations/block_to_block.asm:48]: we still want to run halo2 proofs for these tests to test permutations between machines

File: pipeline/src/test_util.rs [hunk 5]
```diff
@@ -265,7 +265,14 @@ pub fn gen_halo2_proof(pipeline: Pipeline<Bn254Field>, backend: BackendVariant)
     // Setup
     let output_dir = pipeline.output_dir().clone().unwrap();
     let setup_file_path = output_dir.join("params.bin");
-    let max_degree = pil.degrees().into_iter().max().unwrap();
+    let max_degree = pil
+        .degree_ranges()
+        .into_iter()
+        .map(|range| range.max)
+        .collect::<Vec<_>>()
+        .into_iter()
```
Comment (chriseth) [inline pipeline/src/test_util.rs:273]:
```suggestion
```
Does this not work?

Comment (leonardoalt) [reply pipeline/src/test_util.rs:273]: you pasted the same code

File: test_data/asm/permutations/block_to_block.asm (hunk 3 above)
Comment (chriseth) [reply test_data/asm/permutations/block_to_block.asm:48]: But do we test the previous functionality somewhere else?

File: pipeline/src/test_util.rs (hunk 5 above)
Comment (leonardoalt) [reply pipeline/src/test_util.rs:273]: ah read it wrong, the suggestion is only removal, let me check

File: test_data/asm/permutations/block_to_block.asm (hunk 3 above)
Comment (leonardoalt) [reply test_data/asm/permutations/block_to_block.asm:48]: language/compiler/witgen wise yes, the RISCV machine has this stuff and more

File: pipeline/src/test_util.rs (hunk 5 above)
Comment (leonardoalt) [reply pipeline/src/test_util.rs]: it works, changed

**Review by chriseth — APPROVED** [review]

**Pull request merged**